src/sentry/utils/sdk.py

```diff
@@ -158,22 +147,15 @@ def capture_event(event):
             #     event.setdefault('tags', {})['install-id'] = install_id
             upstream_transport.capture_event(event)
 
-        if relay_transport:
-            rate = options.get("store.use-relay-dsn-sample-rate")
-            if rate and random.random() < rate:
-                # Record this before calling `is_current_event_safe` to make
-                # numbers comparable to InternalTransport
-                metrics.incr("internal.captured.events.relay")
-                if is_current_event_safe():
-                    relay_transport.capture_event(event)
-                else:
-                    metrics.incr("internal.uncaptured.events.relay", skip_internal=False)
-                    sdk_logger.warn("internal-error.unsafe-stacktrace.relay")
-                return
-
-        if internal_transport:
-            metrics.incr("internal.captured.events.internal")
-            internal_transport.capture_event(event)
+        if relay_transport and options.get("store.use-relay-dsn-sample-rate") == 1:
+            # Record this before calling `is_current_event_safe` to make
+            # numbers comparable to InternalTransport
```

Comment:
Since InternalTransport is going away, we can probably unify how we record "captured" and "uncaptured" metric.

Reply:
yes you're right